### PR TITLE
VB Generator Fixes

### PR DESCRIPTION
**Pull request by 333fred**
Generate VB code in a platform-agnostic manner that does not introduce cr's on non-windows platforms.

### CODE REVIEW

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a review of this small change.

File: src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/RedNodes/SyntaxFactsWriter.vb [hunk 1]
```diff
@@ -86,6 +86,10 @@ Public Class SyntaxFactsWriter
                     tokenText = tokenText.Replace("""", """""")
                 End If
 
+                If tokenText IsNot Nothing AndAlso tokenText.Contains(vbCrLf) Then
```
Comment (333fred) [inline src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/RedNodes/SyntaxFactsWriter.vb:89]: `LineContinuationTrivia` encodes a CRLF into it's `TokenText`. We replace this with whatever the environment newline is.

**Comment by RikkiGibson** [comment]
Should there be some assertion that the generated code does not introduce CR characters if it is not running on Windows?

**Comment by 333fred** [comment]
> Should there be some assertion that the generated code does not introduce CR characters if it is not running on Windows?

In principle, yes. However, unless we actually run the generators on a Linux/Mac run, realistically you or I are going to be the ones that hit it later down the line, and it's already visible as git diff changes when those files aren't touched.

File: src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/RedNodes/SyntaxFactsWriter.vb (hunk 1 above)
Comment (sharwell) [inline src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/RedNodes/SyntaxFactsWriter.vb:89]:
```suggestion
                If tokenText IsNot Nothing Then
```

Comment (333fred) [reply src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/RedNodes/SyntaxFactsWriter.vb:89]: I'd prefer to keep this as is. I don't see any advantage to normalizing when there's nothing to normalize.

Comment (sharwell) [reply src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/RedNodes/SyntaxFactsWriter.vb:89]: The current approach is running the `Contains` check twice. This one is an unnecessary duplicate.

Comment (333fred) [reply src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/RedNodes/SyntaxFactsWriter.vb:89]: I'd still prefer to keep it as is.

**Comment by 333fred** [comment]
@RikkiGibson any further comments?

**Comment by RikkiGibson** [comment]
> However, unless we actually run the generators on a Linux/Mac run, realistically you or I are going to be the ones that hit it later down the line, and it's already visible as git diff changes when those files aren't touched.

Got it, I guess we only run the generators in the correctness run currently?

**Comment by 333fred** [comment]
> Got it, I guess we only run the generators in the correctness run currently?

Yep.

**Pull request merged**